source/documentation/using_the_api/register_components.md

```diff
@@ -1,18 +1,30 @@
-## The relationship between entries, items and records
+## The components of a register
 
-Registers are made up of:
+Registers are primarily composed of:
 
 * entries
 * items
 * records
 
+### Entries
+
 Entries contain metadata about what and when data has changed. 
 
-Each entry in a register has a relationship with one item, which contains data. An item is connected to an entry with an item hash.   
+Entries are made up of the fields `entry-number`, `index-entry-number`, `entry-timestamp`, `item-hash` and `key`. 
+
+The `entry-number` is unique and defines an entry’s position within the ordered list of a register. The `index-entry-number` is unique and defines an entry’s position within the ordered list of an index. For any given entry in a register, the `entry-number` and `index-entry-number` are always identical. 
+
+The `entry-timestamp` is the time when an entry was introduced to a given register, but it is no guarantee of the order of entries in a register. The `entry-number` is.
+
+### Items
+
+Each entry in a register has a relationship with one item. Items contain data for a given register. Entries and items are connected by a given `item-hash`.  
```

Comment:
I wonder if we could flip the sentences so the definition of what is an item comes first. Also, it would be interesting to hint that an item is the recipient for a structured set of data where the fields are pre-defined and consistent within a register.

Reply:
> Items contain data for a given register.

Would it be more useful if we say that an item contains data for a _thing_ in a register? Perhaps I'm overthinking it.

Reply:
I agree with all of these points, to varying extents. Item first is no problem; I just put entry first because it was the biggest. Describing an item as a structured set of data is also no problem. The only thing I'm not going to propose in the next commit will be to say that items contain data for things in registers, because I've reworked the definition around the first point:

> Items contain structured data sets. They have predefined fields which are consistent within a register.

I think I like this definition because it strays away from an abstract definition of what items are and sticks to a rigorous technical definition. 

That being said if it gets tested and people struggle with it then of course ...

Reply:
I like the reworked definition, let's see how it tests out.

Just to clarify, when I said "flip the sentences so the definition of what is an item comes first" I was talking about the paragraph within the "Items" section, not at all referring to flipping entry and item sections.

Reply:
@arnau Ah, I see. Yes, that makes sense: the definition of an item before was cyclical, and relied on entries to make sense. That wasn't ideal; I think it's better now. 

@nacnudus How about "items contain data for a given entry in a register"?

Reply:
@bahmady That's fine with me :)